src/core/algorithms/md/md_verifier/md_verifier.cpp

```diff
@@ -0,0 +1,223 @@
+
+        default:
+            throw std::runtime_error(
+                    "Failed to calcutate similarity measure: unsupported similarity measure type "
+                    "provided");
+    }
+}
+
+bool MDVerifier::CheckRows(size_t first_row, size_t second_row) {
+    for (std::size_t i = 0; i < lhs_indices_.size(); ++i) {
+        auto index = lhs_indices_[i];
+        auto decision_boundary = lhs_desicion_bondaries_[i];
+        auto similarity_measure = lhs_similarity_measures_[i];
+
+        auto const& column = relation_->GetColumnData(index);
+        auto similarity =
+                CalculateSimilarity(column.GetValue(first_row), column.GetValue(second_row),
+                                    column.GetTypeId(), similarity_measure);
+        if (similarity < decision_boundary) {
+            return true;
+        }
+    }
+    bool holds_for_rhs = true;
+    for (std::size_t i = 0; i < rhs_indices_.size(); ++i) {
+        auto index = rhs_indices_[i];
+        auto decision_boundary = rhs_desicion_bondaries_[i];
+        auto similarity_measure = rhs_similarity_measures_[i];
+
+        auto const& column = relation_->GetColumnData(index);
+        auto similarity =
+                CalculateSimilarity(column.GetValue(first_row), column.GetValue(second_row),
+                                    column.GetTypeId(), similarity_measure);
+        if (similarity < decision_boundary) {
+            holds_for_rhs = false;
+            highlights_.AddHighlight({std::make_pair(first_row, second_row), index,
+                                      column.GetDataAsString(first_row),
+                                      column.GetDataAsString(second_row), similarity,
+                                      decision_boundary});
+            rhs_suggestion_boundaries_[i] = std::min(rhs_suggestion_boundaries_[i], similarity);
+        }
+    }
+    return holds_for_rhs;
+}
+
+void MDVerifier::VerifyMD() {
+    if (lhs_indices_.size() != lhs_desicion_bondaries_.size() ||
+        lhs_indices_.size() != lhs_similarity_measures_.size()) {
+        throw std::length_error(
+                "Length of LHS Indices doesn't match length of LHS Decision Boundaries or/and LHS "
+                "Similarity Measures.");
+    }
+
+    if (rhs_indices_.size() != rhs_desicion_bondaries_.size() ||
+        rhs_indices_.size() != rhs_similarity_measures_.size()) {
+        throw std::length_error(
+                "Length of RHS Indices doesn't match length of RHS Decision Boundaries or/and RHS "
+                "Similarity Measures.");
+    }
+
+    md_holds_ = true;
+
+    rhs_suggestion_boundaries_ = rhs_desicion_bondaries_;
+
+    auto num_cols = relation_->GetNumRows();
+
+    for (size_t first_row = 0; first_row < num_cols; ++first_row) {
+        for (size_t second_row = first_row + 1; second_row < num_cols; ++second_row) {
+            if (!CheckRows(first_row, second_row)) {
+                md_holds_ = false;
+            }
+        }
+    }
```

Comment:
This is extremely simple, it would be preferable to use something like what HyMD does, with some potential for code reuse.